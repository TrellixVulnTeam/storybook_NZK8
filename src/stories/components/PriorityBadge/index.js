// @flow

import React from 'react';
import Badge from 'components/Badge';

type Props = {
  priority: 'NONE' | 'LOW' | 'MEDIUM' | 'HIGH',
  displayStyle?: 'compact' | 'full',
};

export default (props: Props) => {
  const priorityColor = {
    NONE: 'gray',
    LOW: 'blue',
    MEDIUM: 'yellow',
    HIGH: 'red',
  };

  const priorityLabel = {
    NONE: { compact: '', full: 'Unprioritized' },
    LOW: { compact: 'Low', full: 'Low Priority' },
    MEDIUM: { compact: 'Med', full: 'Medium Priority' },
    HIGH: { compact: 'High', full: 'High Priority' },
  };

  return (
    <Badge theme={priorityColor[props.priority]}>
      {priorityLabel[props.priority || 'NONE'][props.displayStyle || 'compact']}
    </Badge>
  );
};
