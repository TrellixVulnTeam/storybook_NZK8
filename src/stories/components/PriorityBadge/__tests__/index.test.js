import React from 'react';
import { shallow } from 'enzyme';
import PriorityBadge from '../index';

describe('<PriorityBadge />', () => {
  describe('compact view', () => {
    const displayStyle = 'compact';

    it('should render correctly with NONE priority with default view', () => {
      const c = shallow(<PriorityBadge priority="NONE" />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with NONE priority', () => {
      const c = shallow(<PriorityBadge priority="NONE" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with LOW priority', () => {
      const c = shallow(<PriorityBadge priority="LOW" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with MEDIUM priority', () => {
      const c = shallow(<PriorityBadge priority="MEDIUM" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with HIGH priority', () => {
      const c = shallow(<PriorityBadge priority="HIGH" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });
  });

  describe('full view', () => {
    const displayStyle = 'full';

    it('should render correctly with NONE priority', () => {
      const c = shallow(<PriorityBadge priority="NONE" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with LOW priority', () => {
      const c = shallow(<PriorityBadge priority="LOW" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with MEDIUM priority', () => {
      const c = shallow(<PriorityBadge priority="MEDIUM" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });

    it('should render correctly with HIGH priority', () => {
      const c = shallow(<PriorityBadge priority="HIGH" displayStyle={displayStyle} />);

      expect(c).toMatchSnapshot();
    });
  });
});
