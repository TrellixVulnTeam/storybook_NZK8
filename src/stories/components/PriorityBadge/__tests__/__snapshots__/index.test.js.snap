// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriorityBadge /> compact view should render correctly with HIGH priority 1`] = `
<_default
  theme="red"
>
  High
</_default>
`;

exports[`<PriorityBadge /> compact view should render correctly with LOW priority 1`] = `
<_default
  theme="blue"
>
  Low
</_default>
`;

exports[`<PriorityBadge /> compact view should render correctly with MEDIUM priority 1`] = `
<_default
  theme="yellow"
>
  Med
</_default>
`;

exports[`<PriorityBadge /> compact view should render correctly with NONE priority 1`] = `
<_default
  theme="gray"
/>
`;

exports[`<PriorityBadge /> compact view should render correctly with NONE priority with default view 1`] = `
<_default
  theme="gray"
/>
`;

exports[`<PriorityBadge /> full view should render correctly with HIGH priority 1`] = `
<_default
  theme="red"
>
  High Priority
</_default>
`;

exports[`<PriorityBadge /> full view should render correctly with LOW priority 1`] = `
<_default
  theme="blue"
>
  Low Priority
</_default>
`;

exports[`<PriorityBadge /> full view should render correctly with MEDIUM priority 1`] = `
<_default
  theme="yellow"
>
  Medium Priority
</_default>
`;

exports[`<PriorityBadge /> full view should render correctly with NONE priority 1`] = `
<_default
  theme="gray"
>
  Unprioritized
</_default>
`;
